tests/serverpod_test_server/lib/src/endpoints/test_tools.dart

```diff
@@ -108,3 +108,63 @@ class AuthenticatedTestToolsEndpoint extends Endpoint {
     }
   }
 }
+
+class TestToolsWithArgVariationsEndpoint extends Endpoint {
```

Comment:
Discussion: Would it make sense to add all these permutations once and then test them both form the "real" generated client in e2e tests and also the test framework?

This will help us validate we actually have client and test framework support for these?

Reply:
Sounds like a good idea, will investigate!

Reply:
Made an attempt here, let me know what you think of its placement/naming: 556ae12